src/network/network.cc

```diff
@@ -442,7 +442,12 @@ string Connection::recv( void )
 	   || (e.the_errno == EWOULDBLOCK) ) {
 	assert( !islast );
 	continue;
-      } else {
+      } else if (e.the_errno == ENOTCONN) {
```

Comment:
Several problems here:
* This fails Travis CI on Linux
* This will break servers, if they ever receive `ENOTCONN`-- the server only has one socket, and `hop_port()` is not appropriate
* This uses the iterator `it` after it's been invalidated by `.erase()`
* If the failing socket is the first one in the container, `it--` moves the iterator past the beginning of the container